[compiler] Force the output in fixed mode for double

Otherwise, they can be truncated and lead to incorrect values
---
 src/compiler/ability.cc |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/compiler/ability.cc b/src/compiler/ability.cc
index c87b1d6..1f92140 100644
--- src/compiler/ability.cc
+++ src/compiler/ability.cc
@@ -144,7 +144,8 @@ struct print_initializer_helpers : boost::static_visitor<std::string>
 	std::string operator () (const Constant<T>& c) const
 	{
 		std::ostringstream oss;
-		oss << ", " << var_name << "(" << c.value << ")";
+		oss.precision(9);
+		oss << ", " << var_name << "(" << std::fixed << c.value << ")";
 		return oss.str();
 	}
 
